src/rpc/methods/eth.rs

```diff
@@ -948,6 +949,73 @@ pub fn new_eth_tx_from_signed_message<DB: Blockstore>(
     Ok(ApiEthTx { hash, ..tx })
 }
 
+/// Creates an Ethereum transaction from Filecoin message lookup. If a negative `tx_index` is passed
+/// into the function, it looks up the transaction index of the message in the tipset, otherwise
+/// it uses the `tx_index` passed into the function.
+fn new_eth_tx_from_message_lookup<DB: Blockstore>(
+    ctx: &Ctx<DB>,
+    message_lookup: &MessageLookup,
+    mut tx_index: i64,
```

Comment:
nit: it's best to do it the Rust way - instead of passing a negative index - just use an `Option` and an unsigned integer instead.

Reply:
Good idea.